### PR TITLE
--pytest-with-coverage should be passed to colcon test, not colcon build

### PR DESCRIPTION
**Pull request by ivanpauno**
Fixes https://github.com/ros2/ci/issues/372.

I need somebody pushing the branch to this repo, in order to be able to run a ci build testing this change (or write access here).

### CODE REVIEW

File: ros2_batch_job/__main__.py [hunk 1]
```diff
@@ -358,13 +357,20 @@ def build_and_test(args, job):
...
         '--build-base', '"%s"' % args.buildspace,
         '--install-base', '"%s"' % args.installspace,
-    ] + (['--merge-install'] if not args.isolated else []) + args.test_args,
-        exit_on_error=False, shell=True)
+    ]
+    if not args.isolated:
+        test_cmd.append('--merge-install')
+    if args.coverage:
+        test_cmd.append('--pytest-with-coverage')
+    test_cmd.append(args.test_args)
```
Comment (dirk-thomas) [inline ros2_batch_job/__main__.py]: This should be `test_cmd += args.test_args` instead.

**Comment by dirk-thomas** [comment]
Thank you for fixing this!

File: ros2_batch_job/__main__.py (hunk 1 above)
Comment (ivanpauno) [reply ros2_batch_job/__main__.py]: Ups, done in https://github.com/ros2/ci/pull/373/commits/04cf53f52ac97936b09a2caa6f866ec44bda7aa1.

**Comment by ivanpauno** [comment]
It's working now:

Coverage build: [Build Status](https://ci.ros2.org/job/nightly_linux_coverage/739/)

Merging!

**Pull request merged**